### PR TITLE
Allow invoices to be submitted to Xero in approved state

### PR DESCRIPTION
**Pull request by agileware**

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
For history, it was originally not possible to submit in approved state. this is a change from Xero's end. I'm about 50% there for saying we should pass as 'Authorized' now they support that. But perhaps we should just default to AUTHORIZED (& leave the option).

Stylistically I prefer civicrm_api3() than civicrm_api() & having to set 'version' - but that's not a blocker.

**Comment by agileware** [comment]
Hey Eileen

What do we need to do here to get this one committed?

Cheers!

File: CRM/Civixero/Invoice.php [hunk 1]
```diff
@@ -199,6 +199,13 @@ protected function mapToAccounts($invoiceData, $accountsID) {
         $lineItems[$index]['UnitAmount'] = -$lineItem['UnitAmount'];
       }
     }
+    
+    // Get default Invoice status
+    $status = civicrm_api('setting', 'getvalue', array(
```
Comment (eileenmcnaughton) [inline CRM/Civixero/Invoice.php]: Ideally use civicrm_api3()

File: settings/Civixero.setting.php [hunk 2]
```diff
@@ -104,4 +110,20 @@
     ),
     'quick_form_type' => 'Element',
   ),
+  'xero_default_invoice_status' => array(
+    'group_name' => 'Xero Settings',
+    'group' => 'xero',
+    'name' => 'xero_default_invoice_status',
+    'type' => 'Array',
+    'add' => '4.4',
```
Comment (eileenmcnaughton) [inline settings/Civixero.setting.php]: this is misleading - I don't think there is an established alternative - perhaps remove

File: CRM/Civixero/Invoice.php [hunk 3]
```diff
@@ -199,6 +199,13 @@ protected function mapToAccounts($invoiceData, $accountsID) {
...
+        'group' => 'Xero Settings',
```
Comment (eileenmcnaughton) [inline CRM/Civixero/Invoice.php]: I think group is not required any more

**Comment by eileenmcnaughton** [comment]
I think I'm fine with this in principle. I made some minor comments. In particular I think settings.getvalue doesn't take group_name in 4.7

File: settings/Civixero.setting.php (hunk 2 above)
Comment (agileware-fj) [reply settings/Civixero.setting.php]: You the 'add' key is misleading?

Comment (eileenmcnaughton) [reply settings/Civixero.setting.php]: yep - I guess so

**Comment by jusfreeman** [comment]
@eileenmcnaughton I think this PR should be closed, I can't track this back to the original work so we'll open a new one, if it's still required.

Keep smiling! :)

**Comment by eileenmcnaughton** [comment]
OK @jusfreeman will do

**Comment by jusfreeman** [comment]
Ref: CIVIXERO-3

**Pull request closed without merging**